analyzers/tests/SonarAnalyzer.UnitTest/CFG/Roslyn/ControlFlowBranchTest.cs

```diff
@@ -128,5 +141,51 @@ public void Method(bool condition)
             entering.LeavingRegions.Should().BeEmpty();
             entering.FinallyRegions.Should().BeEmpty();
         }
+
+        [TestMethod]
+        public void ValidateOperations()
+        {
+            const string code = @"
+public class Sample
+{
+    int Pow(int num, int exponent)
+    {
+        num = num * Pow(num, exponent - 1);
+        return 42;
+    }
+}";
+            var cfg = TestHelper.CompileCfg(code);
+            var entry = cfg.EntryBlock;
+            var exit = cfg.ExitBlock;
+            var body = cfg.Blocks[1];
+            entry.Kind.Should().Be(BasicBlockKind.Entry);
+            entry.Operations.Should().BeEmpty();
+            exit.Kind.Should().Be(BasicBlockKind.Exit);
+            exit.Operations.Should().BeEmpty();
+            body.Kind.Should().Be(BasicBlockKind.Block);
+            body.Operations.Length.Should().Be(1);
+            body.Operations[0].Syntax.Kind().Should().Be(SyntaxKind.ExpressionStatement);
+            body.BranchValue.Syntax.Kind().Should().Be(SyntaxKind.NumericLiteralExpression);
+            body.OperationsAndBranchValue.Should().OnlyContainInOrder(body.Operations[0], body.BranchValue);
+        }
+
+        [TestMethod]
+        public void ValidateAnonymousFunctionFinder()
```

Comment:
This test should be in `RoslynControlFlowGraphTests`
Let's rename it to `FlowAnonymousFunctionOperations_FindsAll` and add two functions inside.

Reply:
Moving it there introduces some difficulties testing it as an extension method, since I have to call it by FQN to avoid conflicts with _DescendantsAndSelf_ extension method on line 85, but it still works.